srtcore/packetfilter.cpp

```diff
@@ -231,6 +231,9 @@ void PacketFilter::globalInit()
     // Add here builtin packet filters and mark them
     // as builtin. This will disallow users to register
     // external filters with the same name.
+
+    filters["fec"] = new Creator<FECFilterBuiltin>;
```

Comment:
When is this deleted?

Reply:
By `ManagedPtr`. Look at definition for `filters`:
```
    typedef std::map<std::string, ManagedPtr> filters_map_t;
    static filters_map_t filters;
```
I had to make a dedicated pointer wrapper for that purpose only, it's modeled after the old `auto_ptr` implementation. In C++11 `unique_ptr` could replace it.

Reply:
`operator=` of `ManagedPtr` should check `owns` and `delete f` before copying a new value.

```
ManagedPtr a(new Creator<FECFilterBuiltin>);
ManagedPtr b(new Creator<FECFilterBuiltin>);
a = b;
```

```
void copy_internal(const ManagedPtr& other)
{
	other.owns = false;
	f = other.f;
	owns = true;
}

void operator=(const ManagedPtr& other)
{
	copy_internal(other);
}
```

Reply:
Right. This was actually predicted that the copy target is always a freshly created empty `ManagedPtr` so that the assignment into the map item can work.

(Fortunately for the already merged portion of the code, this part is not in use until this PR).